Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -52,12 +54,13 @@ public async Task GetByCityNameAsync_ReturnedWeatherDTO_Success()
 
             _weatherApiServiceMock
                 .Setup(weatherApiService =>
-                    weatherApiService
-                    .GetByCityNameAsync(cityName))
+                    weatherApiService.GetByCityNameAsync
+                    (It.Is<string>(x => x == cityName),
+                     It.IsAny<CancellationToken>()))
                 .ReturnsAsync(weatherApiDto);
 
             // Act
-            var result = await _weatherService.GetByCityNameAsync(cityName);
+            var result = await _weatherService.GetByCityNameAsync(cityName, CancellationToken.None);
```

Comment:
where is case for cancellation? same question for other methods

Reply:
+